packages/core/test/combinator/multicast-test.ts

```diff
@@ -157,15 +162,15 @@ describe('MulticastSource', () => {
 
 describe('MulticastDisposable', () => {
   it('should dispose when sinks reaches zero', () => {
-    const expectedSink = {}
-    const source = {
-      remove (sink) {
+    const expectedSink = sink
+    const source = new class extends MulticastSource<unknown> {
+      remove (sink: Sink<unknown>): 0 {
         eq(expectedSink, sink)
         return 0
-      },
+      }
 
-      dispose () {}
-    }
+      dispose (): void {}
+    }(empty())
```

Comment:
I think it's possible to hoist this anonymous class and the one below to a single class declaration ... probably something like:

```ts
class TestMulticastSource<A> extends MulticastSource<A> {
  expectedSink: Sink<A>
  constructor(expectedSink: Sink<A>) {
    super(empty())
    this.expectedSink = expectedSink
  }
  remove (sink: Sink<A>): number {
    eq(this.expectedSink, sink)
    return super.remove(sink)
  }
  dispose (): void {}
}
```

I don't feel strongly about it, but personally, I do find the anonymous classes to be a bit trickier to read.  I figured I'd offer it up as a possibly alternative for discussion.

Reply:
In favor of hoisting, too.

Reply:
done